File: fix_database.py
#!/usr/bin/env python3
import mysql.connector
from mysql.connector import Error

def fix_database():
    try:
        # Connexion à la base de données
        connection = mysql.connector.connect(
            host='localhost',
            user='root',
            password='admin',
            database='clinique_obstetrique'
        )
        
        cursor = connection.cursor()
        
        print("🔧 Début de la correction de la base de données...")
        
        # 1. Supprimer la table grossesses si elle existe
        print("🗑️ Suppression de la table grossesses...")
        cursor.execute("DROP TABLE IF EXISTS grossesses")
        
        # 2. Vider et corriger la table consultations_prenatales
        print("📋 Correction de la table consultations_prenatales...")
        cursor.execute("DELETE FROM consultations_prenatales")
        cursor.execute("ALTER TABLE consultations_prenatales ADD COLUMN IF NOT EXISTS patiente_id INT")
        cursor.execute("ALTER TABLE consultations_prenatales DROP FOREIGN KEY IF EXISTS consultations_prenatales_ibfk_1")
        cursor.execute("ALTER TABLE consultations_prenatales DROP COLUMN IF EXISTS grossesse_id")
        cursor.execute("ALTER TABLE consultations_prenatales ADD CONSTRAINT IF NOT EXISTS fk_consultations_patiente FOREIGN KEY (patiente_id) REFERENCES patientes(id) ON DELETE CASCADE")
        cursor.execute("ALTER TABLE consultations_prenatales MODIFY COLUMN patiente_id INT NOT NULL")
        
        # 3. Vider et corriger la table accouchements
        print("👶 Correction de la table accouchements...")
        cursor.execute("DELETE FROM accouchements")
        cursor.execute("ALTER TABLE accouchements ADD COLUMN IF NOT EXISTS patiente_id INT")
        cursor.execute("ALTER TABLE accouchements DROP FOREIGN KEY IF EXISTS accouchements_ibfk_1")
        cursor.execute("ALTER TABLE accouchements DROP COLUMN IF EXISTS grossesse_id")
        cursor.execute("ALTER TABLE accouchements ADD CONSTRAINT IF NOT EXISTS fk_accouchements_patiente FOREIGN KEY (patiente_id) REFERENCES patientes(id) ON DELETE CASCADE")
        cursor.execute("ALTER TABLE accouchements MODIFY COLUMN patiente_id INT NOT NULL")
        
        # 4. Supprimer les index liés aux grossesses
        print("🗂️ Nettoyage des index...")
        cursor.execute("DROP INDEX IF EXISTS idx_grossesses_patiente_id ON grossesses")
        cursor.execute("DROP INDEX IF EXISTS idx_consultations_grossesse_id ON consultations_prenatales")
        cursor.execute("DROP INDEX IF EXISTS idx_accouchements_grossesse_id ON accouchements")
        
        # 5. Créer de nouveaux index
        print("📊 Création des nouveaux index...")
        cursor.execute("CREATE INDEX IF NOT EXISTS idx_consultations_patiente_id ON consultations_prenatales(patiente_id)")
        cursor.execute("CREATE INDEX IF NOT EXISTS idx_accouchements_patiente_id ON accouchements(patiente_id)")
        
        # Valider les changements
        connection.commit()
        
        print("✅ Base de données corrigée avec succès !")
        print("📝 Modifications effectuées :")
        print("   - Table 'grossesses' supprimée")
        print("   - Table 'consultations_prenatales' : grossesse_id → patiente_id")
        print("   - Table 'accouchements' : grossesse_id → patiente_id")
        print("   - Index mis à jour")
        
    except Error as e:
        print(f"❌ Erreur : {e}")
    finally:
        if connection.is_connected():
            cursor.close()
            connection.close()
            print("🔌 Connexion fermée")

if __name__ == "__main__":
    fix_database() 

File: fix_database.php
<?php
require_once __DIR__ . '/src/Config/Database.php';

use Clinique\Config\Database;

$db = Database::getInstance();

echo "🔧 Correction de la structure de la base de données...\n";

try {
    // 1. Vérifier si la colonne patiente_id existe déjà
    $result = $db->fetchAll("SHOW COLUMNS FROM consultations_prenatales LIKE 'patiente_id'");
    
    if (empty($result)) {
        echo "➕ Ajout de la colonne patiente_id...\n";
        $db->query("ALTER TABLE consultations_prenatales ADD COLUMN patiente_id INT AFTER id");
    } else {
        echo "✅ Colonne patiente_id existe déjà\n";
    }
    
    // 2. Vérifier si la colonne grossesse_id existe
    $result = $db->fetchAll("SHOW COLUMNS FROM consultations_prenatales LIKE 'grossesse_id'");
    
    if (!empty($result)) {
        echo "🗑️ Suppression de la contrainte de clé étrangère grossesse_id...\n";
        try {
            $db->query("ALTER TABLE consultations_prenatales DROP FOREIGN KEY consultations_prenatales_ibfk_1");
        } catch (Exception $e) {
            echo "⚠️ Contrainte déjà supprimée ou nom différent\n";
        }
        
        echo "🗑️ Suppression de la colonne grossesse_id...\n";
        $db->query("ALTER TABLE consultations_prenatales DROP COLUMN grossesse_id");
    } else {
        echo "✅ Colonne grossesse_id n'existe pas\n";
    }
    
    // 3. Ajouter la contrainte de clé étrangère pour patiente_id
    echo "🔗 Ajout de la contrainte de clé étrangère patiente_id...\n";
    try {
        $db->query("ALTER TABLE consultations_prenatales ADD CONSTRAINT fk_consultations_patiente 
                   FOREIGN KEY (patiente_id) REFERENCES patientes(id) ON DELETE CASCADE");
    } catch (Exception $e) {
        echo "⚠️ Contrainte déjà existante\n";
    }
    
    // 4. Créer un index sur patiente_id
    echo "📊 Création de l'index sur patiente_id...\n";
    try {
        $db->query("CREATE INDEX idx_consultations_patiente_id ON consultations_prenatales(patiente_id)");
    } catch (Exception $e) {
        echo "⚠️ Index déjà existant\n";
    }
    
    // 5. Vérifier la structure finale
    echo "\n📋 Structure finale de la table consultations_prenatales:\n";
    $columns = $db->fetchAll("DESCRIBE consultations_prenatales");
    foreach ($columns as $column) {
        echo "  - {$column['Field']}: {$column['Type']} " . 
             ($column['Null'] === 'NO' ? 'NOT NULL' : 'NULL') . 
             ($column['Key'] ? " ({$column['Key']})" : "") . "\n";
    }
    
    echo "\n✅ Correction terminée avec succès!\n";
    
} catch (Exception $e) {
    echo "❌ Erreur: " . $e->getMessage() . "\n";
}
?> 